code/Belgium/Belgium-2020-03-31.py

```diff
@@ -23,8 +25,10 @@
 day_of_march_ICU  = day_of_march
 ICU =             [ 24,  33,  53,  79, 100, 130,  164,  238,  290,  322,  381,  474,  605,  690,  789,  867,  927, 1021]
 # This included 94 retro-active deaths over period before 30 March
-# FIXME; redisitribute given https://twitter.com/brechtdv/status/1244920276070080513
+# redisitribute given https://twitter.com/brechtdv/status/1244920276070080513
 deceased =        [  4,   4,   5,  10,  14,  21,   37,   67,   75,   88,  122,  178,  220,  289,  353,  431,  513,  705] 
+redistribute =    [  2,   0,   0,   2,   2,   3,    5,    6,    8,    9,    7,   11,   10,    9,   20,   66,   32,    0]
+deceased = list(map(add, deceased, accumulate(redistribute)))
```

Comment:
I am confused that the total of redistribution is not 94. Could you investigate please. Thx :-)
Also validate the actual `deceased` list; I fear it will be non-correct ?

```
>>> red = [  2,   0,   0,   2,   2,   3,    5,    6,    8,    9,    7,   11,   10,    9,   20,   66,   32,    0]
>>> from itertools import accumulate
>>> list(accumulate(red))
[2, 2, 2, 4, 6, 9, 14, 20, 28, 37, 44, 55, 65, 74, 94, 160, 192, 192]
```

Reply:
You're right, I included both newly added and retrospective.

Reply:
I'll see what I can do tomorrow.
The data where I wanted to start from (2020-03-20) doesn't conform to the graph in the tweet.
This is where I got:
# data from 2020-03-20:
deceased =        [  4,   4,   5,  10,  14,  21,   37,   67,   75,   88,  122,  178,  220,  289,  353,  431,  513, 513]
and graph from https://twitter.com/brechtdv/status/1244920276070080513:
redistribute =    [  0,   2,   0,   0,   2,   2,   3,    5,    6,    8,    9,    7,   11,   10,    9,   20,   66,   32]

combined:
from operator import add
from itertools import accumulate

result = list(map(add, deceased, accumulate(redistribute)))
becomes:
[4, 6, 7, 12, 18, 27, 46, 81, 95, 116, 159, 222, 275, 354, 427, 525, 673, 705]

According to the graph, only 32 (705 - 673) cases are counted for 2020-03-30.

I'll need to do some more investigations.

Reply:
Thanks :-)
Largely offline since yesterday evening for personal reasons.

I presume the 66 and 32 can be rejected, because they where not "retro-active data" (but fresh data) in the snapshot of 30/31 March.

Reply:
Thanks @jan-veldeman  for your input.

Finally I have some time to investigate.

I think I understand now. The corrections itself are probably the corrections to a _cumulative_value. So, the corrections themselves are also cumulative _already_

So your original proposed code 
`deceased = list(map(add, deceased, accumulate(redistribute)))`
would probably need to be changed to:
`deceased = list(map(add, deceased, redistribute))` [UNTESTED]

I will fix it manually by simply adding the numbers manually, because simpler to see.

Reply:
No problem, hectic times for everybody. Hope all is fine.

The following source of data might be better to start from:
https://epistat.wiv-isp.be/covid/

I'll need a bit more time, but the following snippet would cache the mortality data locally and print out the cumulative sum:

<pre>
import os
from urllib.request import urlretrieve
import json
import pandas as pd

mortality_url = 'https://epistat.sciensano.be/Data/COVID19BE_MORT.json'
mortality_cache = os.path.basename(mortality_url)

if not os.path.exists(mortality_cache):
  urlretrieve(mortality_url, mortality_cache)


with open(mortality_cache, encoding='utf-8') as infile:
  deceased = pd.read_json(infile, convert_dates='DATE')
deceased = deceased.groupby('DATE').sum() # ignore regions
deceased = deceased.cumsum()
print(deceased)
</pre>

Reply:
Thanks, the fix with removing the accumulate seems to work. I entered a commit straight in master, applying essentially your code here, since this PR was closed ...

Reply:
Interesting concept in your comment above.

One important question I see ... I would prefer to have date-time stamped files, so we can always what "data version" was downloaded at which time (theoretically, we would like a copy-on-write optimization for that, but that is premature optimization at this extremely small-data level).

Reply:
Indeed: I was thinking of:
* use a local cache file based on the current date
* in case it doesn't exist or no data was found inside corresponding to the current date: download it (again)

Will not be for today though :-(